File: leetcode/Day01_1446. 连续字符/stringSameLenth.py
class Solution:
    def maxPower(self, s: str) -> int:
        n = len(s)
        maxLen = 1
        left = 0
        right = 0
        while right < n:
            while right < n and s[left] == s[right]:
                right += 1
            maxLen = max(maxLen,right -left)
            left = right
        return maxLen

File: leetcode/Day01_1446. 连续字符/stringSameLenth.cpp
class Solution {
public:
    int maxPower(string s) {
        char tmp = ' ';
        int maxCnt=0,cnt=0;
        for(auto i:s)
        {
            if(i != tmp)
            {
                cnt = 0;
                tmp = i;
            }
            cnt++;
            maxCnt = max(maxCnt,cnt);
        }
        return maxCnt;

    }
};